pkg/apiserver/experiment/experiment.go

```diff
@@ -67,11 +71,50 @@ func (s *Service) listExperiments(c *gin.Context) {}
 // TODO: need to be implemented
 func (s *Service) getExperimentDetail(c *gin.Context) {}
 
-// TODO: need to be implemented
-func (s *Service) deleteExperiment(c *gin.Context) {}
+func (s *Service) deleteExperiment(c *gin.Context) {
+	kind := c.Query("kind")
+	if kind == "" {
+		util.JSONEmptyData(c, statuscode.ParamNotSet, "kind param should be set")
+		return
+	}
+
+	ctx := context.TODO()
+	chaosKey := types.NamespacedName{Namespace: c.Param("ns"), Name: c.Param("name")}
+
+	kindsMap := map[string]runtime.Object{
+		v1alpha1.KindPodChaos:     &v1alpha1.PodChaos{},
+		v1alpha1.KindNetworkChaos: &v1alpha1.NetworkChaos{},
+		v1alpha1.KindStressChaos:  &v1alpha1.StressChaos{},
+		v1alpha1.KindIOChaos:      &v1alpha1.IoChaos{},
+		v1alpha1.KindKernelChaos:  &v1alpha1.KernelChaos{},
+		v1alpha1.KindTimeChaos:    &v1alpha1.TimeChaos{},
+	}
+	var (
+		chaos runtime.Object
+		ok    bool
+	)
+	if chaos, ok = kindsMap[kind]; !ok {
+		util.JSONEmptyData(c, statuscode.UndefinedKind,
+			fmt.Sprintf("kind %s is not supported", kind))
+		return
+	}
+	if err := s.kubeCli.Get(ctx, chaosKey, chaos); err != nil {
+		util.JSONEmptyData(c, statuscode.GetResourcesWrong,
+			"failed to get chaos; "+err.Error())
+		return
+	}
+
+	if err := s.kubeCli.Delete(ctx, chaos, &client.DeleteOptions{}); err != nil {
```

Comment:
I think we need to handle the case that the error is not found. if error is not found, I think we can return success with a business error code?

Reply:
You are right! Forget this case